js/index.js

```diff
@@ -43,41 +43,109 @@ function renderPepperoni() {
 }
 
 function renderMushrooms() {
-  // Iteration 1: set the visibility of `<section class="mushroom">`
+    // Iteration 1: set the visibility of 
+  //`<section class="mushroom">`
+  document.querySelectorAll('.mushroom').forEach(oneMushroom => {
+    if (state.mushrooms) {
+      oneMushroom.style.visibility = 'visible';
+    } else {
+      oneMushroom.style.visibility = 'hidden';
+    }
+  });
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll(".green-pepper").forEach(oneGreenPepper => {
+    if (state.greenPeppers) {
+      oneGreenPepper.style.visibility ="visible";
+    } else {
+      oneGreenPepper.style.visibility = "hidden";
+    }
+  });
 }
 
 function renderWhiteSauce() {
-  // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+//   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  const sauce = document.querySelector(".sauce");  
+  if (state.whiteSauce) {
+    sauce.classList.add('sauce-white'); 
+    } else {
+      sauce.classList.remove('sauce-white'); ;
+     }
 }
 
+// Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
 function renderGlutenFreeCrust() {
-  // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+  const crust = document.querySelector(".crust");
+  if (state.glutenFreeCrust) {
+    crust.classList.add('crust-gluten-free');
+    } else {
+      crust.classList.remove('crust-gluten-free'); 
+  //[0].//crust.removeAttribute('crust-gluten-free');  //I don't understand why removeAttribute doesn't work...
+      }
 }
 
+// Iteration 3: add/remove the class "active" of each `<button class="btn">`
+//1) access .btn (a nodelist therefore can be treated as array)
+//2) forEach ingredientItem (ie element within button array/nodelist), do:
+//3) classList.remove() IF 'active'
+//4) ELSE classList.add() IF !'active
+
+
 function renderButtons() {
-  // Iteration 3: add/remove the class "active" of each `<button class="btn">`
-}
+  let buttonArray = document.querySelectorAll('.btn');
+  // ALTERNATIVE: let buttonArray = document.getElementsByClassName('active');
+  //console.log(buttonArray);
+  for (let i = 0; i < buttonArray.length; i++) {
+  //console.log(buttonArray[i]);
+    if (buttonArray[i].hasAttribute('active') === true) {
+      console.log(383);
+       buttonActive.classList.remove('active');
+      } else buttonArray.classList.add('active');
+      
+    }
+  }
+
+
 
 function renderPrice() {
   // Iteration 4: change the HTML of `<aside class="panel price">`
-}
+  }
 
-renderEverything();
+renderEverything(); {
+
+}
```

Comment:
For this function you can take a look at Jan solution:

function renderPrice() {
// Iteration 4: change the HTML of


let price = basePrice;
let list = '';
for (let ingredient in state) {
if (state[ingredient]) {
list +=
$ ${ingredients[ingredient].price} ${ingredients[ingredient].name}
;
price += ingredients[ingredient].price;
}
}
document.querySelector('.panel.price strong').innerText = $${price};
document.querySelector('.panel.price ul').innerHTML = list;
}